controllers/storagecluster/cephconfig.go

```diff
@@ -151,5 +157,17 @@ func getRookCephConfig(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (s
 		}
 		return rookConfigData, nil
 	}
+	// additonal rook configs when it is a provider cluster
+	if sc.Spec.AllowRemoteStorageConsumers {
+		var rookConfigData string
+		var err error
+		for key, value := range extraRookConfigs {
+			rookConfigData, err = updateRookConfig(defaultRookConfigData, globalSectionKey, key, value)
+			if err != nil {
+				return "", fmt.Errorf("failed to set extra configuration for rook key: %v value: %v | %v", key, value, err)
+			}
+		}
+		return rookConfigData, nil
+	}
```

Comment:
If we start using dual-stack on AWS we will never get to this code, we need a code change that will allow doing each one separately or both depending on the conditions.

Reply:
Added a function to check for AllowRemoteStorageConsumers and append configs accordingly for both cases